app/models/queue_item.rb

```diff
@@ -0,0 +1,26 @@
+class QueueItem < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :video
+
+  delegate :category, to: :video
+  delegate :title, to: :video, prefix: :video
+
+  before_create :set_order
+
+  default_scope { order(:order) }
```

Comment:
Agreed, already experienced some of those unintended side effects. I will probably remove this soon and work with a normal scope.